Src/Base/AMReX_TinyProfiler.cpp

```diff
@@ -92,12 +92,18 @@ TinyProfiler::start () noexcept
         ttstack.emplace_back(std::make_tuple(t, 0.0, &fname));
         global_depth = ttstack.size();
 
+#ifdef AMREX_USE_GPU
+            if (device_synchronize_around_region) {
+                amrex::Gpu::Device::synchronize();
+            }
+#endif
+
 #ifdef AMREX_USE_CUDA
-        if (device_synchronize_around_region) {
-            amrex::Gpu::Device::synchronize();
-        }
         nvtxRangePush(fname.c_str());
 #endif
+#if defined(AMREX_USE_HIP) && defined(AMREX_USE_ROCTX)
```

Comment:
Weiqun's suggestion above:
```suggestion
#elif defined(AMREX_USE_HIP) && defined(AMREX_USE_ROCTX)
```